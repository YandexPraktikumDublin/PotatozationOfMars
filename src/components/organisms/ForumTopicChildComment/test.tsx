import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { ForumTopicChildComment } from '.'

describe('<ForumTopicChildComment />', () => {
  const childComment = {
    id: 2,
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    userId: 1,
    user: {
      id: 1,
      login: 'IvanIvanov',
      name: 'Ivan Ivanov',
      createdAt: '01.12.20',
      updatedAt: '01.12.20'
    },
    topicId: 1,
    parentId: 1,
    hierarchyLevel: 1,
    reactions: [
      {
        id: 2,
        content: 'ðŸ˜‚',
        userId: 1,
        user: {
          id: 1,
          login: 'IvanIvanov',
          name: 'Ivan Ivanov',
          createdAt: '01.12.20',
          updatedAt: '01.12.20'
        },
        commentId: 2,
        createdAt: '01.12.20',
        updatedAt: '01.12.20'
      }
    ],
    createdAt: '01.12.20',
    updatedAt: '01.12.20'
  }

  it('should renders correct <ForumTopicChildComment />', () => {
    const wrapper = shallow(
      <ForumTopicChildComment childComment={childComment} />
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
