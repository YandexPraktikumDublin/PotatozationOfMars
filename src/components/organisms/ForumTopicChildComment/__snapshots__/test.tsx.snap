// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForumTopicChildComment /> should renders correct <ForumTopicChildComment /> 1`] = `
<div
  className="text-left border border-primary rounded-2xl p-4 mb-4 ml-4 last:mb-0 dark:border-white"
>
  <div
    className="flex justify-between mb-2 text-xs"
  >
    Ivan Ivanov
     at 
    1/12/2020, 12:00 AM
  </div>
  <div
    className="mb-2"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </div>
  <ForumTopicCommentActions
    comment={
      Object {
        "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        "createdAt": "01.12.20",
        "hierarchyLevel": 1,
        "id": 2,
        "parentId": 1,
        "reactions": Array [
          Object {
            "commentId": 2,
            "content": "ðŸ˜‚",
            "createdAt": "01.12.20",
            "id": 2,
            "updatedAt": "01.12.20",
            "user": Object {
              "createdAt": "01.12.20",
              "id": 1,
              "login": "IvanIvanov",
              "name": "Ivan Ivanov",
              "updatedAt": "01.12.20",
            },
            "userId": 1,
          },
        ],
        "topicId": 1,
        "updatedAt": "01.12.20",
        "user": Object {
          "createdAt": "01.12.20",
          "id": 1,
          "login": "IvanIvanov",
          "name": "Ivan Ivanov",
          "updatedAt": "01.12.20",
        },
        "userId": 1,
      }
    }
  />
</div>
`;
